File: web/src/app/domain/documentation.ts
import { IPageble } from '../core/pagings';
import { BaseEntity } from './base-entity';

export class Documentation extends BaseEntity implements IPageble {
  splitBy: string = '<hr>';

  source(): string {
    return this.content;
  }

  area!: DocumentArea;
  title!: string;
  content!: string;
}

export enum DocumentArea {
  Admin,
  Teacher,
  Student,
}

File: Api/src/Domain/Entities/Documentation.cs
namespace Domain.Entities
{
    public class Documentation : IEntity
    {
        public Guid Id { get; set; }
        public DocumentArea Area { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
    }

    public enum DocumentArea
    {
        Admin,
        Teacher,
        Student
    }
}
